### ISSUE TITLE
Add official geometries for Finnish localities

### ISSUE THREAD

**Issue by hannesj** [whosonfirst-data/whosonfirst-data#99]
The current geometries for Finnish localities coming from Quattroshapes are quite sparse.
The official geometries for Finnish municipalities (=localities) can be found at [here](http://kartat.kapsi.fi/files/kuntajako/kuntajako_10k/etrs89/gml/TietoaKuntajaosta_2015_10k.zip) and are licensed as CC4.0-BY by the National Land Survey of Finland.

**Comment by hannesj** [comment]
Also, the official gazetteer for all Finnish place names (in all five languages) can be found [here](http://kartat.kapsi.fi/files/nimisto/paikat/etrs89/gml/paikat_2015_05.zip)

**Comment by nvkelso** [comment]
Thanks! We'll have a look :)

**Comment by hannesj** [comment]
We would be interested in contributing to this. What would be the best steps to proceed?

**Comment by nvkelso** [comment]
Hi @hannesj, sorry for the long delay getting back to you. Is there a SHP or GeoJSON version of this file available? 

Next step is we need to figure out the Who's On First `id`in the existing data compared to the new data. That can be accomplished with a point-in-polygon operation and then manual verification of the [728 existing locality](https://whosonfirst.mapzen.com/spelunker/id/85633143/descendants/?&placetype=locality) records. Once we have the `id` concordance, the new geometries can be attached for the existing records, and the remainder imported. Another option is to deprecate the existing record but mark that it's been superseded by a new version. Either way we need the concordance. Some of the importing is still a manual process that @thisisaaronland would need to be involved in (setting up admin hierarchy, etc).

We already include data from National Land Survey of Finland (via Quattroshapes), and this version is significantly better quality and under the same type of open data license.

**Comment by hannesj** [comment]
Hi.

I created individual WoF-formatted GeoJSON-files out of the bigger file, cut those along the coastline and mapped to existing WoF-records. However, I realised that possibly these should be imported as localadmin rather than localities, since even tough most of them cover localities 1 to 1, some of them are covering multiple (existing) localities. What is your opinion on this?

I uploaded the files to [here](http://koti.kapsi.fi/~hannes/finland.zip). They need to be merged with the existing data for the localities/localadmins/regions and metadata refreshed. What would be the best path to progress with these?

**Comment by nvkelso** [comment]
Thanks! Back at you with a few questions. It'll take me a little bit to research to form an opinion on the 1st (below).

**Questions:**

1. It does sound like localadmin versus localities at first glance, but I need to research. Are the WOF `id` on the filenames assuming locality or localadmin or mixed?
2. Is this download only features with WOF concordances? How many other features from the source should we add?
3. Can we also get the original geoms as well as the coastline clipped geoms? We'd put the larger geoms into an alt geometry file (useful for reverse geocoding applications).
4. Mapzen can take care of adding all the other hierarchy bits when we ingest the final files. @thisisaaronland has made good progress on that this week.

**Comment by hannesj** [comment]
1. After looking at other countries' data in WoF and QS, I'm turning into thinking that these are actually localadmins. The question is, what should we do with bigger cities where locality matches exactly with localadmin? The ids are the localities that are the "administrative seats" for the localareas. Also, records starting with `4042273` are macroregions and those starting with `85683` regions which should get new geometries (and proper translations). Records starting with `85667` are municipalities in Åland which should be changed to be of the same placetype as the other records (either locality or localadmin).
2. This is all geometries definied in the xml-file. There is only one record, `au351122.geojson`, which didn't have any concordance to WoF, so I left it named as it was in the original source, even tough it corresponds to a locality or localadmin. 
3. They can be found converted to GeoJSON from [here](http://koti.kapsi.fi/~hannes/SuomenKuntajako_2015_10k.geojson). These can be matched using the nlsfi:nationalCode property.
4. Thanks, please let me know if something comes up or how I can help further.

**Comment by hannesj** [comment]
@nvkelso Is there something in addition we could do to help with this?

**Comment by nvkelso** [comment]
I've been swamped with a few deadlines elsewhere. I can have a look Monday.

**Comment by hannesj** [comment]
Sorry for the bump. Is there any way we could help with this?

**Comment by hannesj** [comment]
@nvkelso, Sorry to bother you, but is there anything we could do to help this to go through?

**Comment by vesameskanen** [comment]
I am also available, if there's something I can do to speed up the process. We just got a customized Pelias WOF admin mapping working very nicely, and it would be great to get the actual geographic data updated as well. Currently the results are poor, so this has a very high priority for us.

**Comment by nvkelso** [comment]
Over to @stepps00 for integration.

**Comment by nvkelso** [comment]
##### `2ndOrder` features (`macroregion`)

@hannesj and @vesameskanen Your [file](http://koti.kapsi.fi/%7Ehannes/SuomenKuntajako_2015_10k.geojson) includes shapes for the 7 old provinces (`2ndOrder`), which roughly line up with what WOF has as `macroregion` now. But our research shows those were abolished in 2010, being replaced with the 19 provinces (`3rdOrder`) you include. 

  1. Does anyone use the old 2nd order features anymore? 
  2. I propose deprecating those WOF `macroregion` features and not importing your `2ndOrder` features at all, do you agree?

##### `3rdOrder` features (`region`)

For the `3rdOrder` features, propose importing those are WOF `region`. But we only have 35, so likely that's pre reorganization and we should just deprecate our old WOF records, import new WOF features from yours.

##### `4thOrder` features (`localadmin`)

For the `4thOrder` features, agree on importing those as WOF `localadmin` as entirely new features.

##### locality

No shape change proposed at this time, but will recalculate hierarchy based on above.

##### neighbourhood

No shape change proposed at this time, but will recalculate hierarchy based on above.

**Comment by hannesj** [comment]
I updated the sources for this year to [here](http://koti.kapsi.fi/~hannes/SuomenKuntajako_2016_10k.geojson.zip). This was due to two merged localadmins. The data is still provided as CC-BY-4.0 [Maanmittauslaitos/National Land Survey of Finland](http://www.maanmittauslaitos.fi/en/digituotteet/municipal-division-finland)

As for your questions:

### 2ndOrder (`macroregion`)

These map to the [these](https://whosonfirst.mapzen.com/spelunker/search/?q=finland&placetype=macroregion#5/64.430/28.476) current `macroregion` types (+`dependency` [Åland](https://whosonfirst.mapzen.com/spelunker/id/85632789/#7/60.173/20.215)) and are in use today. Before 2010 we had [six provinces](https://en.wikipedia.org/wiki/Provinces_of_Finland) which were replaced by [the current areas](https://en.wikipedia.org/wiki/Regional_State_Administrative_Agency) pretty much 1:1, only Southwestern Finland was split off. I propose that we just change the geometries and names of the existing six macroregions and Åland.

### 3ndOrder (`region`)

These are the 19 [regions](https://en.wikipedia.org/wiki/Regions_of_Finland), that map cleanly to the current 19 [regions in WoF](https://whosonfirst.mapzen.com/spelunker/id/85633143/descendants/?placetype=region&iso=fi#5/64.211/23.843). Similarly as above, the geometries and names should be synced for these.

For some reason Åland has its municipalities currently marked as [regions](https://whosonfirst.mapzen.com/spelunker/id/85633143/descendants/?placetype=region&iso=ax#9/60.1871/21.0202) in WoF. These should probably be deprecated and be recreated as localadmins from `4thOrder`. The situation of Åland is a bit challenging as it is represented by a single entity on multiple governmental levels.

### 4thOrder (`localadmin`)

These map cleanly to the `localadmin` type and should be imported as such.

### `locality`

I found the probable origin for the Quattroshapes localities. [These](http://koti.kapsi.fi/~hannes/Taajamat_2014.geojson.zip) are the localities provided by  [Suomen Ympäristökeskus/Finnish Environment Institute](http://metatieto.ymparisto.fi:8080/geoportal/catalog/search/resource/details.page?uuid=%7BCB02E608-6E6B-417E-87DA-BD7C8CB53712%7D) as CC-BY-4.0. Unfortunately these are not cut by the localadmins, so that contiguous populated areas are represented by a single locality named after the their central place, i.e. [Vantaa](https://whosonfirst.mapzen.com/spelunker/id/101748419/#11/60.3058/24.9688) and [Espoo](https://whosonfirst.mapzen.com/spelunker/id/101748415/#10/60.2104/24.6748) are included in the locality of Helsinki, So I don't know if these could be used. There are not however the metro regions, which are provided in a separate database, but just contiguous built-up areas. These provide the population info in the field `Vaesto14Lk`.

### `postalcode`

[These](https://whosonfirst.mapzen.com/spelunker/id/85633143/descendants/?placetype=postalcode&page=2#14/0.0000/0.0000) records should be updated with the proper geometry, which is provided by [Tilastokeskus/Statiscs Finland](http://www.stat.fi/tup/paavo/index_en.html) as CC-BY-4.0. The data can be found [here](http://koti.kapsi.fi/~hannes/Paavo_2016.geojson.zip). The data has the field `POSTI_ALUE`, which contains the postal code, `NIMI` and `NAMN` which have the Finnish and Swedish name for the post office (a.k.a postal city) and `HE_VAKIY` which has the population living in that postal code area

### `neighbourhood`

Unfortunately no common neighbourhood info exists for the whole country, and not that many records exist in WoF [today](https://whosonfirst.mapzen.com/spelunker/id/85633143/descendants/?placetype=neighbourhood&page=1#6/62.246/24.941). However, the records in Helsinki are a bit [wonky](https://whosonfirst.mapzen.com/spelunker/id/85907887/#12/60.1454/24.9612) and could be replaced with the official data for [Helsinki](http://koti.kapsi.fi/~hannes/Neighborhoods_Helsinki.geojson.zip) which is provided by [the city](http://www.hri.fi/en/dataset/helsingin-kaupunginosat) under CC-BY-4.0.

**Comment by nvkelso** [comment]
Very helpful, thanks!

**Comment by stepps00** [comment]
Thank you for the detail, @hannesj!

After review, it seems the **2ndOrder** (`macroregion`), and **3rdOrder** (`region`) data will be used for `geometry` and `wof:name` updates. I agree with your assertion of the **4thOrder** data being used as new (`localadmin`) records for Finland.

I'm not certain that the [locality](http://koti.kapsi.fi/~hannes/Taajamat_2014.geojson.zip) data from the [Suomen Ympäristökeskus/Finnish Environment Institute](http://metatieto.ymparisto.fi:8080/geoportal/catalog/search/resource/details.page?uuid=%7BCB02E608-6E6B-417E-87DA-BD7C8CB53712%7D) is the source of the Quattroshapes data after comapring the two datasets. That being said, the Suomen Ympäristökeskus/Finnish Environment Institute shapes do cover a somewhat larger geography than our current Finnish localities, so I will look into importing. Similarly, I will take a look at the Helsinki neighbourhood data and postal code data.

Thanks again!

**Comment by vesameskanen** [comment]
A quick note: current WOF data of Finland excludes Swedish versions of names. This is unfortunate, because Swedish is one of the official languages here. Most data sources @hannesj listed above seem to include all language versions, and it would be great to have them included by this update.

**Comment by stepps00** [comment]
Hi @vesameskanen! 

As you mentioned, our branch contains [new fields](https://github.com/whosonfirst/whosonfirst-data/commit/5ec44d0cd06edef0fa76949ce28e44ad7213f50c) for the Finnish `macroregion` records. These fields `name:lang_x_variant` and `name:lang_x_preferred` fields for Finnish, English, and Swedish. A goal is to add the Swedish and FInnish name fields to the `region` and `localadmin` level records, as well. 

https://en.wikipedia.org/wiki/Regions_of_Finland
https://en.wikipedia.org/wiki/List_of_Finnish_municipalities

**Comment by stepps00** [comment]
The `localadmin` records are now in the S3 WOF bucket.

**Comment by vesameskanen** [comment]
This is great news, many thanks!

**Comment by hannesj** [comment]
@stepps00 Thanks! Are you planning to push these to git or how will the process continue?

**Comment by stepps00** [comment]
@vesameskanen @hannesj No problem! Yes, these new records and updated geometries will make their way to our WOF repository, though we are still updating/creating new records for the smaller geographies (`locality`, `neighbourhood`, `postalcode`). Keep an eye on this issue thread and Issue #247 for updates to Finland.    :-)

**Comment by thisisaaronland** [comment]
Yeah, this is on me. It's near the top of the pile but is waiting on a couple other things. Soon soon...

**Comment by stepps00** [comment]
@hannesj A question about Helsinki's `microhoods` for you.

On this [page on the subdivisions of Helsinki](https://en.wikipedia.org/wiki/Subdivisions_of_Helsinki#Neighbourhoods), 59 neighbourhoods are listed and map cleanly to the neighbourhoods .geojson provided in the comments above. However, you'll also notice sub-divisions under several of the neighbourhoods listed. For instance, under `11. Kallio (Berghäll)`, the following are listed:

   * `111 Siltasaari (Broholmen)`
   * `112 Linjat (Linjerna)`
   * `113 Torkkelinmäki (Torkelsbacken)`

Are  `Siltasaari`, `Linjat`, and `Torkkelinmäki` commonly referred to as microhoods in the `Kallio` neighbourhood? We're updating the neighbourhood records in Helsinki and want to ensure the new records' hierarchies are correct, as there are various geographic divisions in Helsinki (ex. [peruspiiri districts](https://en.wikipedia.org/wiki/Subdivisions_of_Helsinki#Districts)).

**Comment by hannesj** [comment]
Uh, oh... It's complicated...

Probably I'd rather make the smaller areas `neighbourhood`s and the 59 "neighborhoods" `macrohood`s. Some of these are the same and for those, only the `neighborhood` should be created. The geometries for the smaller units can be found [here](https://gist.github.com/hannesj/d9fd4fa6147e49629da2644347d34431). There are some (non-official) areas smaller than these (especially in the city) for which I would like to keep the `microhood` available for.

**Comment by hannesj** [comment]
[Here](https://gist.github.com/hannesj/0fa9130ce830eed8dd888ccd671d7723) is the smallest division. Unfortunately these don't have names included, but they conform to the regularly used `microhood`s. Eg. Viiskulma has the id of `112531` and Erottaja `112650`

**Comment by stepps00** [comment]
Thank you for the information! I think you're correct when you suggested making the smaller areas `neighbourhoods` and making the 59 "neighborhoods" `macrohoods`. Will investigate the smaller divisions.

**Comment by stepps00** [comment]
@hannesj Another question about Finnish administrative records for you..

The screenshot below shows our current WOF records (transparent with a black boundary) and the new locality records you provided (light red shading). You'll notice the black labels show the current `wof:name` in our records' data and the red labels that show the new locality records' `TaajNimi` name field. 

Are these new `TaajNimi` name fields more appropriate for locality names or are the current `wof:name` labels more accurate? What is the difference between these two name fields? 

Examples:

WOF Name vs. TaajNimi field:
`Loimaa` vs. `Loimaan kt.` 
`Aura` vs. `Auran as.seutu` 
`Tarvasjoki` vs. `Tarvasjoen kk.`

<img width="672" alt="screen shot 2016-06-02 at 3 09 43 pm" src="https://cloud.githubusercontent.com/assets/18567700/15762710/274e2a26-28d4-11e6-8f34-515e833120fd.png">

**Comment by hannesj** [comment]
Hi,

The names provided in `TaajNimi`are more formal than the existing names. They should possibly be added as alternatives, but the existing names should probably be kept.

The -n in the name is just the genitive suffix, meaning Loimaa's, Aura's, Tarvasjoki's, etc.

The other suffices are as follows.

- kt. [Keskustaajama](https://fi.wikipedia.org/wiki/Keskustaajama) = Chef-lieu or Administrative centre
- kk. [Kirkonkylä](https://fi.wikipedia.org/wiki/Kirkonkyl%C3%A4) = Parish(?) i.e. a village with its own church. Might also mean a place differing from the old village centre if the church was built outside the village at the time when it was built. Typically these have evolved to be the central location for the village.
- as.seutu [Asemanseutu](https://fi.wikipedia.org/wiki/Asemanseutu) = Station village(?) These were created when the railroads were built so that they did not pass close by to the church in the village or the station was built outside the existing city. The areas around the stations have usually grown at a faster pace than the old villages.

An example of this is in Loimaa [kirkonkylä](http://www.openstreetmap.org/way/255469920#map=14/60.8753/22.9837) [asemanseutu](http://www.openstreetmap.org/node/380670896#map=14/60.8493/23.0624), together forming the keskustaajama.

Hoping this makes sense.

**Comment by stepps00** [comment]
@hannesj Again, thank you for clarifying; your explanation makes sense.
This likely means we will import the new `TaajNimi` name as a `fin_x_variant` name field, keep the existing `wof:name` fields, and update geometries for our locality records. 

Thanks!

**Comment by nvkelso** [comment]
@thisisaaronland has some S3 work to do here, and @stepps00 has some further work around the localities. 

(In this case, the new Github assignment feature is useful, lol.)

**Comment by vesameskanen** [comment]
Any estimate when the changes will be finished? I'd like to get started implementing a translation feature for Pelias geocoding, which uses WOF for translating admin parts of addresses into  local languages. So any proceedings to get Finnish WOF data updated would be greatly appreciated. Many thanks in advance for your valuable work!

**Comment by thisisaaronland** [comment]
My hope is sometime this week or next. Specifically, we are using issue #300 to kick the tires on automating the kinds of cascading updates these sorts of changes yield. This in turn has meant dealing with issue #338 which is basically done.

Apologies for the delay. I am the bottleneck right now and some days this is fighting for attention with all the other things. Working on this stuff is the focus for the week.

**Comment by thisisaaronland** [comment]
Just a quick note to say: This has finally bubbled up to the top of my list. I am waiting on some updated data files from @stepps00 and then I will start working on it.

**Comment by stepps00** [comment]
The S3 misc bucket now has an [Issue 99](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%2099) folder that contains geojson files for import. The files are the following:

**Issue99_Fin_localadmin_import.geojson**

* Brand new localadmin geometries to import. Whos on First does not know about any of these records, they should all be imported with new `wof:ids`.

**Issue99_Fin_Macroregion_import.geojson**

* A collection of macroregion records for Finland. Includes a `wof:id` field that should map to an existing macroregion record, a `wof:name` field that should be used to replace the existing `wof:name`, and a preferred `name` field for both Swedish and Finnish.

**Issue99_Fin_Region_import.geojson**

* A collection of region records for Finland. Includes a `wof:id` field that should map to an existing region record, a `wof:name` field that should be used to replace the existing `wof:name`, and a preferred `name` field for both Swedish and Finnish.

**Issue99_Fin_Postcode_import.geojson** and **Issue99_Fin_Postcode_imports_2.geojson**

* These postcode geometries and attributes have already been imported into Whos on First.

Some work still needs to be done to map new locality geometries to existing Whos on First locality records in Finland. Once complete, I'll update this Issue with a new comment.

**Comment by vesameskanen** [comment]
Many thanks for these improvements! Can you give a rough estimate when the new data will become available via  WOF git repository?

**Comment by thisisaaronland** [comment]
Neighbourhoods and macrohoods have been imported. Unless there is an unforeseen problem with the import chain then macroregions and regions will happen today.

That leaves a couple placetypes; @stepps00 has the details on those.

**Comment by stepps00** [comment]
Hi @vesameskanen!

After the macroregion and region import today, we will have imported the following data in our WoF repository:

* Macroregions in Finland
* Regions in Finland
* Postalcodes in Finland
* Localadmins (the municipality shapes that were the original request for this issue) in Finland
* Boroughs in Helsinki
* Macrohoods in Helsinki

This leaves potential imports for Helsinki neighbourhood and Finnish [locality](http://koti.kapsi.fi/~hannes/Taajamat_2014.geojson.zip) records, if they are deemed as improvements to our current WoF records (the locality data is similar to what we currently have in WoF).

Since we are currently juggling Finland updates in two separate issues, I suggest we close both issues this week and open a new issue for any unforseen problems/new issues.

**Comment by vesameskanen** [comment]
Thanks for the quick reply. Importing proper locality records has a top priority for us, because we use WOF data for a geocoding application and the locality name is a standard part of Finnish addresses.

**Comment by stepps00** [comment]
No problem, @vesameskanen. I believe the `locality` shapes you are referring to were determined to be `localadmin` shapes that @hannesj discusses above (the **4thOrder** shapes). These `localadmin` shapes are contiguous and snapped to one another, while the `locality` shapes are quite sparse.

**Comment by thisisaaronland** [comment]
Localadmins are being imported in dry-run mode as I write this. Assuming no bad craziness they'll be part of `whosonfirst-data` shortly.

**Comment by thisisaaronland** [comment]
Where "bad craziness" means ensuring that the meta files are updated before restarting the PIP servers in order to rebuild the hierarchy for descendants of a given placetype – so I am doing that now.

Note: A consequence of this is that rebuilding the hierarchy for descendants of macroregion and region was technically incorrect so there _may_ be errors. My hunch is that there aren't but I will re-enhierarchy-ize all the things once localadmins have been sorted out.

**Comment by vesameskanen** [comment]
Million thanks for your attention and super fast feedback! I will test this as the first thing tomorrow morning.

**Comment by thisisaaronland** [comment]
Sigh... this is blocked on https://github.com/whosonfirst/py-mapzen-whosonfirst-meta/issues/3

**Comment by thisisaaronland** [comment]
Things are chugging along. Most things seem correct, when rebuilding hierarchies with the exception of known weird geometries for some boroughs like these:

https://whosonfirst.mapzen.com/spelunker/id/890536883/
https://whosonfirst.mapzen.com/spelunker/id/890536879/

The simplest, dumbest thing to do for these is just to manually set the `lbl:` centroid as a separate update (commit) so that the PIP server doesn't return an empty array for possible parents (locality).

The right to do is to update the polygons to not contain... all that ice water. Which, I don't know... maybe? I'm from Canada so if you told me that (icy waters) was a Finnish thing I would believe you :-)

Either way, we're getting there. This has all been super helpful from a building-tools perspective.

**Comment by stepps00** [comment]
Our `borough` and `macrohood` source shapes in Helsinki, as well as `localadmin` source shapes for Finland cover areas of water in the Baltic Sea... Once imported, I can clip these new administrative boundaries to the country shape for Finland and store the clipped (or original) shapes as alt-geometries.

**Comment by hannesj** [comment]
@thisisaaronland I'm not sure the neighborhoods were yet updated. At least the spelunker still contains the following erreneous ones.

https://whosonfirst.mapzen.com/spelunker/id/85907891/#15/60.1633/24.9383
The name `Katu` is `Street` in finnish, and should probably just be deperecated.

https://whosonfirst.mapzen.com/spelunker/id/85898897/#14/60.1577/24.9375
`Suomenlinna` is a group of islands south of the city, but the geometry seems to contain most of Punavuori

https://whosonfirst.mapzen.com/spelunker/id/85907887/#13/60.1525/24.9745
`Punavuori` in it's place contains parts of the correct neightborhood but also parts of Suomenlinna and Ullanlinna.

**Comment by hannesj** [comment]
@stepps00 There is a problem with that, as the Finnish [shape](https://whosonfirst.mapzen.com/spelunker/id/85633143/#12/60.1482/24.9798) doesn't seem to cover any islands. [Example 2](https://whosonfirst.mapzen.com/spelunker/id/85633143/#10/60.3473/21.4522). I posted a nice geometry covering the whole country, but unfortunately it's several hundreds of MBs.

**Comment by hugovk** [comment]
> https://whosonfirst.mapzen.com/spelunker/id/85907891/#15/60.1633/24.9383
> The name Katu is Street in finnish, and should probably just be deperecated.
> 
> https://whosonfirst.mapzen.com/spelunker/id/85898897/#14/60.1577/24.9375
> Suomenlinna is a group of islands south of the city, but the geometry seems to contain most of Punavuori
> 
> https://whosonfirst.mapzen.com/spelunker/id/85907887/#13/60.1525/24.9745
> Punavuori in it's place contains parts of the correct neightborhood but also parts of Suomenlinna and Ullanlinna.

I think these may derive from Flickr shapefiles, which were generated using geotagged photos and can be a good rough guide, but are sometimes a bit iffy. At least there's Katu and Suomenlina (the real island-fortress has two Ls: Suomenlinna).

https://code.flickr.net/2008/11/20/boundaries-a-tool-to-explore-flickrs-shapefiles/
http://boundaries.tomtaylor.co.uk/#23701410

![image](https://cloud.githubusercontent.com/assets/1324225/17515427/f18a511a-5e40-11e6-888a-4ddbff704ca1.png)

**Comment by thisisaaronland** [comment]
@hannesj - Neighbourhoods have not been imported yet, that is correct. I will let @stepps00 speak to the details about that.

As an aside, the work is being divided up between the two of us in such a way that I mostly deal with "finished" data sets specifically so that we can design and build tools in as abstract and generic a way as possible rather than each import being a special snowflake. So far everything is still a snowflake but that's the work, anyway... :-) 

@hugovk – It will be 10 years since we launched geotagging at the end of the month!

**Comment by stepps00** [comment]
Hey, @hannesj - that's correct, the `neighbourhood` shapes in Helsinki are one of the remaining pieces for import. We have focused our neighbourhood update efforts on other locations, but reconciling new data with our existing Who's on First records in Helsinki is on the to-do list.

We've also developed a handy [neighbourhood update guide](https://github.com/whosonfirst/whosonfirst-docs/wiki/Issue-%23125---A-Guide-to-Updating-Neighbourhood-Records-in-Who's-on-First-%28part-1%29) in case anyone outside of Mapzen has neighbourhood contributions!

**Comment by thisisaaronland** [comment]
@stepps00 - localadmins have been imported. Would you please create a new issue for the remaining placetypes.

**Comment by stepps00** [comment]
@thisisaaronland - separate issues have been opened and referenced above. I will close this issue, unless we're missing something.

**Comment by hannesj** [comment]
Thank you fot all the work!

@thisisaaronland For some reason this didn't add the new localadmins to the hierarchies of the localities etc?

**Comment by vesameskanen** [comment]
Big thanks indeed, now we got the long-awaited localamin data! 

For your information, I noticed some errors when fetching finnish WOF data using the go-whosonfirst-clone tool:

getting localadmin data
[wof-clone-metafiles] 12:07:58.157735 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/195/837/907195837.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.168124 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/195/839/907195839.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.168138 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/195/821/907195821.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.181161 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/195/873/907195873.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.186222 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/196/495/907196495.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.242628 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/195/843/907195843.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.273294 [error] Failed to GET https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/195/945/907195945.geojson, because we expected 200 from source and got '404 Not Found' instead
[wof-clone-metafiles] 12:07:58.486169 [error] Failed to HEAD https://s3.amazonaws.com/whosonfirst.mapzen.com/data/907/198/465/907198465.geojson, because we expected 200 from source and got '500 Internal Server Error' instead
[wof-clone-metafiles] 12:07:58.652603 [info] scheduled: 8 completed: 7 success: 0 error: 7 skipped: 0 to retry: 7 goroutines: 34 time: 1.000049004s

So, part of localadmins are apparently missing.

**Comment by thisisaaronland** [comment]
Hrm, that looks like some leftover cruft from all the dry-runs. Apologies and I will make a fresh bundle shortly (the meta file has been updated already).

```
grep 907195843 meta/wof-localadmin-latest.csv | wc -l
0
```

**Comment by hannesj** [comment]
@thisisaaronland Apparently [this thing](https://github.com/whosonfirst-data/whosonfirst-data/issues/247#issuecomment-219654374) now happened with Pelias. When searching for [Hamarintie, Jyväskylä](http://pelias.io/compare/#/v1/search%3Ftext=Hamarintie,%20Jyv%C3%A4skyl%C3%A4) you get results including Hamarintie, Jyvaskyla in the label, which is just incorrect. 
The result is the same if you would write `San Francisco` as `Zan Francizco` based on the fact that s and z look and sound a bit similar, since a, å and ä are totally different characters in Finnish, which just happen to look a bit similar in the latin alphabet.

There is an issue (pelias/pelias#339) open for this, but it would be very important for us to get this fixed.
